marlowe-playground-client/src/MarloweEditor/State.purs

```diff
@@ -0,0 +1,222 @@
+    pure unit
+
+handleAction settings (SetEditorText contents) = do
+  editorSetValue contents
+  -- FIXME
+  -- updateContractInState contents
+  -- pure unit is only here so the comment stays in place
+  pure unit
+
+handleAction _ (ShowBottomPanel val) = do
+  assign _showBottomPanel val
+  editorResize
+
+handleAction _ (ShowErrorDetail val) =  {- FIXME assign _showErrorDetail val -} pure unit
+
+handleAction _ (ChangeBottomPanelView view) = do
+  assign _bottomPanelView view
+  assign _showBottomPanel true
+  editorResize
+
+handleAction _ SendToSimulator = pure unit
+
+handleAction _ ViewAsBlockly = pure unit
+
+handleAction _ (InitMarloweProject contents) = do
+  editorSetValue contents
+  liftEffect $ LocalStorage.setItem marloweBufferLocalStorageKey contents
+  assign _hasUnsavedChanges' false
+
+handleAction _ MarkProjectAsSaved = assign _hasUnsavedChanges' false
+
+handleAction _ (SelectHole hole) = assign _selectedHole hole
+
+handleAction settings AnalyseContract =
+  void
+    $ runMaybeT do
+        contents <- MaybeT $ editorGetValue
+        contract <- hoistMaybe $ parseContract' contents
+        -- FIXME: when editor and simulator were together the analyse contract could be made
+        --        at any step of the simulator. Now that they are separate, it can only be done
+        --        with initial state
+        let
+          emptySemanticState = emptyState zero
+        assign _analysisState (WarningAnalysis Loading)
+        response <- lift $ checkContractForWarnings contract emptySemanticState
+        assign _analysisState (WarningAnalysis response)
+  where
+  checkContractForWarnings contract state = runAjax $ (flip runReaderT) settings (Server.postMarloweanalysis (MSReq.Request { onlyAssertions: false, contract, state }))
+
+handleAction settings AnalyseReachabilityContract =
+  void
+    $ runMaybeT do
+        contents <- MaybeT $ editorGetValue
+        contract <- hoistMaybe $ parseContract' contents
+        -- FIXME: when editor and simulator were together the analyse contract could be made
+        --        at any step of the simulator. Now that they are separate, it can only be done
+        --        with initial state
+        let
+          emptySemanticState = emptyState zero
+        newReachabilityAnalysisState <- lift $ startReachabilityAnalysis settings contract emptySemanticState
+        assign _analysisState (ReachabilityAnalysis newReachabilityAnalysisState)
+
+handleAction _ Save = pure unit
+
+parseContract' :: String -> Maybe Contract
+parseContract' = fromTerm <=< hush <<< parseContract
+
+runAjax ::
+  forall m a.
+  ExceptT AjaxError (HalogenM State Action ChildSlots Void m) a ->
+  HalogenM State Action ChildSlots Void m (WebData a)
+runAjax action = RemoteData.fromEither <$> runExceptT action
+
+editorResize :: forall state action msg m. HalogenM state action ChildSlots msg m Unit
+editorResize = void $ query _marloweEditorPageSlot unit (Monaco.Resize unit)
```

Comment:
@krisajenkins is there some `query_` or something, I can't find it anywhere but it feels like it should exist (possibly with a better name since "query" suggests returning something)